core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -1211,6 +1211,24 @@ private void checkColumnUniquenessForFilterWithConstantColumns(String sql) {
         .assertThatUniqueKeysAre(bitSetOf());
   }
 
+  @Test void testColumnUniquenessForSubqueryWithCorrelatingVars() {
+    final String sql = ""
+        + "select empno, deptno, slacker\n"
+        + "from emp A\n"
+        + "where empno = (\n"
+        + "  select max(empno)\n"
+        + "  from emp B\n"
+        + "  where A.deptno = B.deptno\n"
+        + ")";
+    sql(sql)
+        .assertThatAreColumnsUnique(bitSetOf(0), is(true))
+        // This requires drilling into the subquery
+//        .assertThatAreColumnsUnique(bitSetOf(1), is(true))
```

Comment:
This test was added to test whether we do not see a scalar subquery as constant if it has correlating variables, which it does not, as designed, however, for more complex reasons that were not addressed in this PR, `deptno` is a unique column.